activitysim/benchmarking/componentwise.py

```diff
@@ -0,0 +1,601 @@
+        pipeline.cleanup_pipeline()
+
+
+########
+
+
+def local_dir():
+    benchmarking_directory = workspace.get_dir()
+    if benchmarking_directory is not None:
+        return benchmarking_directory
+    return os.getcwd()
+
+
+def model_dir(*subdirs):
+    return os.path.join(local_dir(), "models", *subdirs)
+
+
+def template_setup_cache(
+    example_name,
+    component_names,
+    benchmark_settings,
+    benchmark_network_los,
+    config_dirs=("configs",),
+    data_dir="data",
+    output_dir="output",
+    settings_filename="settings.yaml",
+    skip_component_names=None,
+    config_overload_dir="dynamic_configs",
+):
+    """
+
+    Parameters
+    ----------
+    example_name : str
+        The name of the example to benchmark, as used in
+        the `activitysim create` command.
+    component_names : Sequence
+        The names of the model components to be individually
+        benchmarked.  This list does not need to include all
+        the components usually included in the example.
+    benchmark_settings : Mapping
+        Settings values to override from their usual values
+        in the example.
+    benchmark_network_los : Mapping
+        Network LOS values to override from their usual values
+        in the example.
+    config_dirs : Sequence
+    data_dir : str
+    output_dir : str
+    settings_filename : str
+    skip_component_names : Sequence, optional
+        Skip running these components when setting up the
+        benchmarks (i.e. in pre-run).
+    config_overload_dir : str, default 'dynamic_configs'
+    """
+    try:
+        os.makedirs(model_dir(), exist_ok=True)
+        get_example(
+            example_name=example_name,
+            destination=model_dir(),
+            benchmarking=True,
+        )
+        os.makedirs(model_dir(example_name, config_overload_dir), exist_ok=True)
+
+        # Find the settings file and extract the complete set of models included
+        from ..core.config import read_settings_file
+
+        try:
+            existing_settings, settings_filenames = read_settings_file(
+                settings_filename,
+                mandatory=True,
+                include_stack=True,
+                configs_dir_list=[model_dir(example_name, c) for c in config_dirs],
+            )
+        except Exception:
+            logger.error(f"os.getcwd:{os.getcwd()}")
+            raise
+        if "models" not in existing_settings:
+            raise ValueError(
+                f"missing list of models from {config_dirs}/{settings_filename}"
+            )
+        models = existing_settings["models"]
+        use_multiprocess = existing_settings.get("multiprocess", False)
+        for k in existing_settings:
+            print(f"existing_settings {k}:", existing_settings[k])
+
+        settings_changes = dict(
+            benchmarking=True,
+            checkpoints=True,
+            trace_hh_id=None,
+            chunk_training_mode="disabled",
```

Comment:
If we have chunking enabled in our settings file, does it get overwritten here?  If so, this needs to be clearly and obviously stated in the main documentation file that chunking is not implemented.

Reply:
Added a warning in the documentation: 
    The benchmarks do not currently use ActivitySim's dynamic chunking features,
    as these require manual configuration and training on a per-machine basis
    to ensure good performance.